### PR TITLE
14/create pre hook

### LINKED ISSUE #14: Update plugin to match with new file system
The upload limiter should maybe be connected with `/download` endpoints instead of `createItem`. 

Currently hitting the limit still result in a success message in builder, but of course the item is not created (but very possibly the file is uploaded).

### PR DESCRIPTION
**Pull request by pyphilia**
This PR completely refactor the plugin, so we can extract a task to be used in graasp-file-item.

close #14

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -65,6 +65,8 @@
       /* Advanced Options */
       "skipLibCheck": true,                     /* Skip type checking of declaration files. */
       "forceConsistentCasingInFileNames": true  /* Disallow inconsistently-cased references to the same file. */
-    }
+    },
+    "include":["src"],
+    "exclude":["*.test.ts"]
   }
```
Comment (juancarlosfarah) [inline tsconfig.json]: Fix end of file.

File: src/tasks/base-task.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  data: Partial<IndividualResultType<R>>;
+  preHookHandler?: PreHookHandlerType<R>;
+  postHookHandler?: PostHookHandlerType<R>;
+
+  getInput?: () => unknown;
+  getResult?: () => unknown;
+
+  constructor(actor: A) {
+    this.actor = actor;
+    this.status = 'NEW';
```
Comment (juancarlosfarah) [inline src/tasks/base-task.ts:30]: Why don't we have these strings factored out again?

**Review by juancarlosfarah — CHANGES_REQUESTED** [review]
Minor issue and a question.

File: src/tasks/base-task.ts (hunk 2 above)
Comment (pyphilia) [reply src/tasks/base-task.ts:30]: These are not factored out in @graasp/types yet so we cannot match the types. At some point it will be unified :/

**Pull request merged**